Schema.rb.example updated

divyabaid16 opened the pull request:
Fixes #0000 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

[//]: # (To mark checkbox write 'x' within the square brackets)

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

jywarren commented:
Ah, i'm sorry, i think due to various PRs, this needs a rebase again? Thank you, @divyabaid16 !

divyabaid16 commented:
@jywarren I have rebased my PR. But I cannot see the Travis build :thinking:

jywarren commented:
Seems like it's working now? Sometimes it takes a little while to start!

jywarren commented:
OK, great, it passed. I'm requesting a couple more reviews just to be sure!

gauravano commented:
Hey @divyabaid16, why are you deleting the migration?

divyabaid16 commented:
I deleted the migration because when it was present and after executing the db setup command, it was showing the error that versions table already exists. Can't be duplicated. So in order to remove that I deleted the migration. Travis was failing.

kaustubh-nair commented:
Hi @divyabaid16 but if you delete the migration, the versions table will not be created while running `rails db:migrate` . Don't you think it's safer to keep the migration file?

divyabaid16 commented:
Ya, that is true.
So let me just add it again.

SidharthBansal commented:
How is it going? Need any help on this?

divyabaid16 commented:
I have done this one. If any change is required I will make it.
Thanks!

alaxalves commented on db/migrate/20190715092943_create_versions.rb:
```diff
@@ -9,7 +9,7 @@ class CreateVersions < ActiveRecord::Migration[5.2]
   TEXT_BYTES = 1_073_741_823
 
   def change
-    create_table :versions, { options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci" } do |t|
```
We need to enforce UTF-8, this is required by Rails 6. I think we should keep this.

sashadev-sky commented:
@alaxalves @divyabaid16 I was just wondering, why do we keep a `schema.rb.example` file if we check in `schema.rb`. Do we need this at all?

alaxalves commented:
> @alaxalves @divyabaid16 I was just wondering, why do we keep a `schema.rb.example` file if we check in `schema.rb`. Do we need this at all?

I have questioned this myself. But I imagined that we keep an example file in case someone wants to "lift" a MK instance. Or if we ourselves maintain a different version of MK. If this is not true we can either erase the `schema.rb` file or the example file

sashadev-sky commented:
I was thinking maybe lets delete it? It's autogenerated. Its almost like checking in a `yarn.lock.example` right? And in that case it would be a 'copy' not so much an example @divyabaid16 what are thoughts

alaxalves commented:
> I was thinking maybe lets delete it? It's autogenerated. Its almost like checking in a `yarn.lock.example` right? And in that case it would be a 'copy' not so much an example @divyabaid16 what are thoughts

Totally agree with this. It's less overhead!

divyabaid16 commented:
Ya, it sounds good to me too.

sashadev-sky commented:
@divyabaid16 ok awesome! So do you want to delete it here? I dont know the other hald of the work you were doing here with versions what was that for?

I could also just delete it in my PR #959

jywarren commented:
Just noting that we do not track it in plots2 -- only the example, in case
people want to build a database without running through the migrations one
by one. In case that helps the discussion!

On Thu, Sep 5, 2019 at 2:44 PM Sasha Boginsky <notifications@github.com>
wrote:

> @divyabaid16 <https://github.com/divyabaid16> ok awesome! So do you want
> to delete it here? I dont know the other hald of the work you were doing
> here with versions what was that for?
>
> I could also just delete it in my PR #959
> <https://github.com/publiclab/mapknitter/pull/959>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/mapknitter/pull/938?email_source=notifications&email_token=AAAF6J4ADMJ7LPP64T43LHDQIFHNJA5CNFSM4ILQYVLKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6AJKIQ#issuecomment-528520482>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6J7XYLCTTK744ZIHZHTQIFHNJANCNFSM4ILQYVLA>
> .
>

sashadev-sky commented:
@jywarren i'm confused, what is the benefit then to checking in an exact duplicate of the `schema.rb` instead of the file itself
